simapp/app.go

```diff
@@ -374,6 +374,10 @@ func NewSimApp(
 		authz.NewAppModule(appCodec, app.AuthzKeeper, app.AccountKeeper, app.BankKeeper, app.interfaceRegistry),
 		transferModule,
 	)
+	// give upgrade keeper the module manager
+	app.UpgradeKeeper.SetModuleManager(app.mm)
+	// pass the updated keeper to the module manager
+	app.mm.Modules[upgradetypes.ModuleName] = upgrade.NewAppModule(app.UpgradeKeeper)
```

Comment:
Do we need this line? Isn't a reference passed?

Reply:
all the keepers passed into module.NewManager are passed by value. I did try to pass upgrade itself by ref but it introduces lots of new errors. I can look into this more if we want to avoid having this extra line here

Reply:
hmm, I removed this line and ran `TestMigrations`, it still passes. Does anything fail when removing this line?